quaireaux/searching/src/binary_search.cairo

```diff
@@ -0,0 +1,41 @@
+use array::SpanTrait;
+use quaireaux_utils::check_gas;
+
+fn binary_search<T,
+impl TCopy: Copy<T>,
+impl TDrop: Drop<T>,
+impl TEq: PartialEq<T>,
+impl TOr: PartialOrd<T>>(
+    span: Span<T>, val: T
+) -> Option<u32> {
+    check_gas();
+
+    // Initial check
+    if (span.len() == 0) {
+        return Option::None(());
+    }
+    let middle = span.len() / 2;
+    if (*span[middle] == val) {
+        return Option::Some(middle);
+    }
+    if (span.len() == 1) {
+        return Option::None(());
+    }
+    if (*span[middle] > val) {
+        return (binary_search(span.slice(0, middle), val));
+    } else {
```

Comment:
Since the previous if is always returning, you don't need the else, this would reduce tabulation ^^'

Reply:
Indeed !